src/dreyfus_index_updater.erl

```diff
@@ -64,10 +70,64 @@ update(IndexPid, Index) ->
         couch_db:close(Db)
     end.
 
-load_docs(FDI, {I, IndexPid, Db, Proc, Total, LastCommitTime}=Acc) ->
+load_docs(FDI, {I, IndexPid, Db, Proc, Total, LastCommitTime, ExcludeIdRevs}=Acc) ->
     couch_task_status:update([{changes_done, I}, {progress, (I * 100) div Total}]),
     DI = couch_doc:to_doc_info(FDI),
-    #doc_info{id=Id, high_seq=Seq, revs=[#rev_info{deleted=Del}|_]} = DI,
+    #doc_info{id=Id, high_seq=Seq, revs=[#rev_info{rev=Rev}|_]} = DI,
+    %check if it is processed in purge_index to avoid update the index again.
+    case lists:member({Id, Rev}, ExcludeIdRevs) of
+        true -> ok;
+        false -> update_or_delete_index(IndexPid, Db, DI, Proc)
+    end,
+    %% Force a commit every minute
+    case timer:now_diff(Now = now(), LastCommitTime) >= 60000000 of
+        true ->
+            ok = clouseau_rpc:commit(IndexPid, Seq),
+            {ok, {I+1, IndexPid, Db, Proc, Total, Now}};
+        false ->
+            {ok, setelement(1, Acc, I+1)}
+    end.
+
+purge_index(Db, IndexPid, Index) ->
```

Comment:
This function is all wrong. There's no reason to run docs through the index function during a purge. This logic should just use fold_purge_infos and then `clouseau_rpc:delete(IndexPid, Id)` for each row. If a document was partially purged it'll be re-inserted when we process the change normally because partially purged documents are always re-inserted into the changes feed.

Reply:
For hastings, the index will be directly deleted/updated in easton while there is cycle about 30 seconds in clouseau side to perform final commit. So we are trying to perform batch processing here to purge indexes. More previous discussion can be found at https://github.com/cloudant-labs/dreyfus/pull/14/files#diff-8b1222f75a3b6a12a1a0ca49c6c5458aR91